dataherald/repositories/golden_records.py

```diff
@@ -0,0 +1,43 @@
+from bson.objectid import ObjectId
+
+from dataherald.types import GoldenRecord
+
+DB_COLLECTION = "golden_records"
+
+
+class GoldenRecordRepository:
+    def __init__(self, storage):
+        self.storage = storage
+
+    def insert(self, golden_record: GoldenRecord) -> GoldenRecord:
+        golden_record.id = self.storage.insert_one(
+            DB_COLLECTION, golden_record.dict(exclude={"id"})
+        )
+        return golden_record
+
+    def find_one(self, query: dict) -> GoldenRecord | None:
+        row = self.storage.find_one(DB_COLLECTION, query)
+        if not row:
+            return None
+        return GoldenRecord(**row)
+
+    def update(self, golden_record: GoldenRecord) -> GoldenRecord:
+        self.storage.update_or_create(
+            DB_COLLECTION,
+            {"_id": ObjectId(golden_record.id)},
+            golden_record.dict(exclude={"id"}),
+        )
+        return golden_record
+
+    def find_by_id(self, id: str) -> GoldenRecord | None:
+        row = self.storage.find_one(DB_COLLECTION, {"_id": ObjectId(id)})
+        if not row:
+            return None
+        return GoldenRecord(**row)
+
+    def find_all(self) -> list[GoldenRecord]:
+        rows = self.storage.find_all(DB_COLLECTION)
+        return [GoldenRecord(**row) for row in rows]
```

Comment:
This is returning id as null
`[
  {
    "id": null,
    "question": "what is my name?",
    "sql_query": "select * 1 from foo",
    "db_alias": "db_alias",
    "created_time": "2023-08-21T15:50:34.923000"
  }
]`
you should set something like this
`return [GoldenRecord(id=row["_id"], **row) for row in rows]
`

Reply:
@jcjc712 Done.